src/machine/machine_atsamd21.go

```diff
@@ -971,69 +1099,69 @@ func setPinCfg(p uint8, val sam.RegValue8) {
 	case 31:
 		sam.PORT.PINCFG0_31 = val
 	case 32: // PB00
-		sam.PORT.PINCFG1_0 = sam.RegValue(val) << 24
+		sam.PORT.PINCFG1_0 |= sam.RegValue(val) << 24
```

Comment:
I wonder whether this does the intended operation when unsetting a value?
The old behavior is probably wrong (overwrites configuration of neighboring pins), but I don't think the new behavior is right. It will likely need some masking, for example:
```go
sam.PORT.PINCFG1_0 = (sam.PORT.PINCFG1_0 & (0xff << 24)) | (sam.RegValue(val) << 24)
```

Reply:
That is a good point, I will make some appropriate changes.

Reply:
I think the correct expression in your example should be:

```go
sam.PORT.PINCFG1_0 = (sam.PORT.PINCFG1_0 &^ (0xff << 24)) | (sam.RegValue(val) << 24)
```

First clear only that part of the register, then set only that part of the register.

Reply:
Yes you are right, I meant the `&^` (and not) operator. I was rather tired when I wrote that comment...

Reply:
OK all corrected.